### PR TITLE
Profile update menu

### PR DESCRIPTION
**Pull request by leddest**
Now a user able to:
- Change full name
- Change email
- Change short user description ("about me")
- Update profile picture

### CODE REVIEW

File: app/database/models.py [hunk 1]
```diff
@@ -11,6 +13,10 @@ class User(Base):
     username = Column(String, unique=True)
     email = Column(String, unique=True)
     password = Column(String)
+    full_name = Column(String)
+    description = Column(String, default="Happy new user!")
```
Comment (ivarshav) [inline app/database/models.py:17]: Maybe set the defauls in a const ?

File: app/routers/profile.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    url = router.url_path_for("profile")
+    response = RedirectResponse(url=url, status_code=HTTP_302_FOUND)
+    return response
+
+
+@router.post("/update_user_email")
+async def update_user_email(
+        request: Request, session=Depends(get_db)):
+
+    user = session.query(User).filter_by(id=1).first()
```
Comment (ivarshav) [inline app/routers/profile.py:82]: Probably should add a TODO to replace the `id=1` part

File: app/routers/profile.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        "user": user,
+        "events": upcouming_events
+    })
+
+
+@router.post("/update_user_fullname")
+async def update_user_fullname(
+        request: Request, session=Depends(get_db)):
+
+    user = session.query(User).filter_by(id=1).first()
```
Comment (efratush) [inline app/routers/profile.py:63]: Filter always id=1 not by user id?

File: schema.md [hunk 4]
```diff
@@ -26,4 +28,6 @@
 ├── schema.md
 └── tests
     ├── __init__.py
+    └── config.py
```
Comment (ivarshav) [inline schema.md:31]: conftest

File: tests/test_profile.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    new_name_data = {
+        'fullname': 'Peter'
+    }
+    # Get profile page and initialize database
+    profile = profile_test_client.get('/profile')
+
+    # Post new data
+    profile = profile_test_client.post(
+        '/profile/update_user_fullname', data=new_name_data)
+    assert profile.status_code == 302
```
Comment (ivarshav) [inline tests/test_profile.py:52]: Can use `from starlette import status`

File: schema.md (hunk 4 above)
Comment (leddest) [reply schema.md:31]: Thank you for paying attention

File: app/routers/profile.py (hunk 3 above)
Comment (leddest) [reply app/routers/profile.py:63]: There's still no user verification

File: app/routers/profile.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+templates = Jinja2Templates(directory="app/templates")
+
+router = APIRouter(
+    prefix="/profile",
+    tags=["profile"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+def get_new_user():
```
Comment (yammesicka) [inline app/routers/profile.py:27]: Please change to "get_placeholder_user"

File: app/templates/base.html [hunk 7]
```diff
@@ -48,5 +50,16 @@
     <script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.9.4/Chart.min.js"
         integrity="sha512-d9xgZrVZpmmQlfonhQUvTR7lMPtO7NkZMkA0ABN3PHCbKA5nqylQ/yWlFAyY6hYgdF1Qh6nYiuADWwKB4C2WSw=="
         crossorigin="anonymous"></script>
+    <script>
```
Comment (yammesicka) [inline app/templates/base.html:53]: Can you please move it to another JS file?

File: app/routers/profile.py [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    finally:
+        session.close()
+
+        url = router.url_path_for("profile")
+        response = RedirectResponse(url=url, status_code=HTTP_302_FOUND)
+        return response
+
+
+async def process_image(image, user):
+    img = Image.open(io.BytesIO(image))
```
Comment (yammesicka) [inline app/routers/profile.py:140]: Can we somehow close this handler?

File: app/routers/profile.py [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+    new_description = data['description']
+
+    # Update database
+    user.description = new_description
+    session.commit()
+
+    session.close()
+
+    url = router.url_path_for("profile")
+    response = RedirectResponse(url=url, status_code=HTTP_302_FOUND)
```
Comment (yammesicka) [inline app/routers/profile.py:112]: You can directly return it

File: app/routers/profile.py [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+    user = session.query(User).filter_by(id=1).first()
+    pic = await file.read()
+
+    try:
+        await process_image(pic, user)
+
+        # Save to database
+        user.avatar = f"{user.username}{PICTURE_EXTENSION}"
+        print(user.avatar)
```
Comment (yammesicka) [inline app/routers/profile.py:128]: Remove the print

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Pull request closed without merging**